### PR TITLE
Fix: automate - Improve yaml loader

### PR DESCRIPTION
**Pull request by oxedions**

### CODE REVIEW

**Comment by strus38** [comment]
Hi
While reviewing this role, I would change the way you are handling python dependencies.
I think it would be better and more flexible to add a requirements.txt file with all python dependencies and as a task, either load the RPMs as you did or enable to install it through pip3. For flask/Celery it is way better to use a venv than using system dependencies.

File: roles/automate/files/bbautomate.py [hunk 1]
```diff
@@ -38,11 +37,20 @@ class bcolors:
 
 def load_file(filename):
     logging.info(bcolors.OKBLUE+'Loading '+filename+bcolors.ENDC)
+
+    # Select YAML loader (needs PyYAML 5.1+ to be safe)
+    yaml_version_major=int(yaml.__version__.split('.')[0])
+    yaml_version_minor=int(yaml.__version__.split('.')[1])
+    if yaml_version_major > 5 or (yaml_version_major == 5 and yaml_version_minor >= 1):
+        yaml_loader='Safe'
```
Comment (neilmunday) [inline roles/automate/files/bbautomate.py:45]: Rather using a hard coded string, why not use a Boolean variable instead? E.g.

```
yaml_loader_safe = False
if yaml_version_major > 5 or (yaml_version_major == 5 and yaml_version_minor >= 1):
    yaml_loader_safe = True
```

File: roles/automate/files/bbautomate.py [hunk 2]
```diff
@@ -38,11 +37,20 @@ class bcolors:
...
+    else:
+        yaml_loader='Unsafe'
+
     with open(filename, 'r') as f:
-        if version.parse(yaml.__version__) < version.parse("5.1.0"):
-            return yaml.load(f)
-        else:
+        if yaml_loader == 'Safe':
```
Comment (neilmunday) [inline roles/automate/files/bbautomate.py:50]: Following on from previous comment:

```
if yaml_loader_safe:
    return yaml.load(f, Loader=yaml.FullLoader)
return yaml.load(f)
```

Note: the `else` is not needed due to the first `return`

**Review by neilmunday — CHANGES_REQUESTED** [review]

**Comment by oxedions** [comment]
@neilmunday Should be better now 🙂

**Comment by oxedions** [comment]
> Hi While reviewing this role, I would change the way you are handling python dependencies. I think it would be better and more flexible to add a requirements.txt file with all python dependencies and as a task, either load the RPMs as you did or enable to install it through pip3. For flask/Celery it is way better to use a venv than using system dependencies.

Ok, so you mean that using a variable (for example `automate_dependencies`), the role would either install packages, or trigger a pip Ansible module (https://docs.ansible.com/ansible/latest/collections/ansible/builtin/pip_module.html) combined with a requirement.txt file to install dependencies via pip ? Do I understand well ?

**Review by neilmunday — APPROVED** [review]

**Comment by oxedions** [comment]
@neilmunday thanks for the review !
@strus38 I will open an issue to continue our exchange.

**Pull request merged**